sdk/storage/azblob/README.md

```diff
@@ -1,443 +1,276 @@
 
-serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+Learn more about options for authentication _(including Connection Strings, Shared Key, Shared Key Signatures, Active Directory, and anonymous public access)_ [in our examples.](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/examples_test.go)
 
-cred, err := service.NewSharedKeyCredential(accountName, accountKey)
-handleError(err)
+### Goroutine safety
+We guarantee that all client instance methods are goroutine-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/golang_introduction.html#thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across goroutines.
 
-serviceClient, err := service.NewClientWithSharedKey(serviceURL, cred, nil)
-handleError(err)
+### About blob metadata
+Blob metadata name/value pairs are valid HTTP headers and should adhere to all restrictions governing HTTP headers. Metadata names must be valid HTTP header names, may contain only ASCII characters, and should be treated as case-insensitive. Base64-encode or URL-encode metadata values containing non-ASCII characters.
 
-fmt.Println(serviceClient.URL())
-```
+### Additional concepts
+<!-- CLIENT COMMON BAR -->
+[Client options](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy#ClientOptions) |
+[Accessing the response](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#WithCaptureResponse) |
+[Handling failures](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore#ResponseError) |
+[Logging](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/log)
+<!-- CLIENT COMMON BAR -->
 
-##### 2. Creating the client from a connection string
+## Examples
 
-You can use connection string, instead of providing the account URL and credential separately, for authentication as
-well. To do this, pass the connection string to the client's `NewServiceClientFromConnectionString` method. The
-connection string can be found in your storage account in the [Azure Portal][azure_portal_account_url] under the "Access
-Keys" section or with the following Azure CLI command:
+### Uploading a blob
 
-```bash
-az storage account show-connection-string -g my-resource-group -n mystorageaccount
-```
+```go
+const (
+	account       = "https://MYSTORAGEACCOUNT.blob.core.windows.net/"
+	containerName = "sample-container"
+	blobName      = "sample-blob"
+	sampleFile    = "path/to/sample/file"
+)
 
-```golang
-import "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+// authenticate with Azure Active Directory
+cred, err := azidentity.NewDefaultAzureCredential(nil)
+// TODO: handle error
 
-connectionString := "DefaultEndpointsProtocol=https;AccountName=<myAccountName>;AccountKey=<myAccountKey>;EndpointSuffix=core.windows.net"
+// create a client for the specified storage account
+client, err := azblob.NewClient(account, cred, nil)
+// TODO: handle error
```

Comment:
For better or worse, I think we've agreed to if(err) {panic(err) } in samples/snippets.

Reply:
I think for examples we should omit it (less ceremony for brevity).  For end-to-end samples I agree that `panic()` is good as you get a stack trace.

Reply:
This is something we should standardize. I think we're doing things differently in this area and it's a little unclear.

Reply:
(I'll file an issue for future discussion)

Reply:
Added a comment about this in https://github.com/Azure/azure-sdk-for-go/issues/18786

Reply:
In fact, I might try migrating this to a `doc.go` file in a separate PR.